### PR TITLE
Issue login view selection replace akita

### PR DESCRIPTION
**Pull request by jeremie74**

### CODE REVIEW

File: apps/catalog/catalog/src/app/marketplace/marketplace.component.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { Component, ChangeDetectionStrategy, OnDestroy } from '@angular/core';
 import { OrganizationQuery } from '@blockframes/organization/+state';
-import { BucketQuery, BucketService } from '@blockframes/contract/bucket/+state';
-import { Subscription, Observable } from 'rxjs';
-import { switchMap, map } from 'rxjs/operators';
+import { BucketService } from '@blockframes/contract/bucket/+state';
+import { Observable } from 'rxjs';
+import { switchMap, map, tap } from 'rxjs/operators';
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/marketplace.component.ts:5]: you can remove tap now

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts [hunk 2]
```diff
@@ -93,7 +92,7 @@ export class MarketplaceMovieAvailsComponent implements AfterViewInit, OnDestroy
 
     const paramsSub = combineLatest([
       this.route.queryParams.pipe(filter(params => !!params.contract && !!params.term)),
-      this.bucketQuery.selectActive().pipe(filter(bucket => !!bucket))
+      this.bucketService.active$.pipe(filter(bucket => !!bucket))
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts:95]: I'm thinking that this should be an issue if there is no bucket for the org yet. Can you try to do a selection with an org that has no bucket yet ?

File: apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts [hunk 3]
```diff
@@ -192,7 +189,8 @@ export class ListComponent implements OnDestroy, OnInit {
     }
 
     const orgId = this.orgQuery.getActiveId();
-    if (this.bucketQuery.getActive()) {
+    const bucket = await this.bucketService.active$.pipe(take(1)).toPromise();
```
Comment (GrandSchtroumpf) [inline apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts:192]: You can use `const bucket = await this.bucketService.getActive()` now

File: libs/contract/src/lib/bucket/+state/bucket.service.ts [hunk 4]
```diff
@@ -2,22 +2,24 @@ import { Injectable } from '@angular/core';
...
+import { shareReplay, switchMap, take } from 'rxjs/operators';
 
 @Injectable({ providedIn: 'root' })
 @CollectionConfig({ path: 'buckets' })
 export class BucketService extends CollectionService<BucketState> {
-
+  active$ = this.orgQuery.selectActiveId().pipe(
+    switchMap((orgId) => this.valueChanges(orgId)),
+    shareReplay(1)
+    );
```
Comment (GrandSchtroumpf) [inline libs/contract/src/lib/bucket/+state/bucket.service.ts:20]: can you outdent it ?

**Review by GrandSchtroumpf — APPROVED** [review]

File: apps/catalog/catalog/src/app/marketplace/marketplace.component.ts (hunk 1 above)
Comment (jeremie74) [reply apps/catalog/catalog/src/app/marketplace/marketplace.component.ts]: Done

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts (hunk 2 above)
Comment (jeremie74) [reply apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts:95]: I tested and it work, without selection and until the end of adding an offer

File: apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts (hunk 3 above)
Comment (jeremie74) [reply apps/catalog/catalog/src/app/marketplace/title/list/list.component.ts]: Done

File: libs/contract/src/lib/bucket/+state/bucket.service.ts (hunk 4 above)
Comment (jeremie74) [reply libs/contract/src/lib/bucket/+state/bucket.service.ts]: Done, if it was what you wanted:
```
active$ = this.orgQuery.selectActiveId().pipe(
    switchMap((orgId) => this.valueChanges(orgId)),
    shareReplay(1)
 );
```

File: apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts (hunk 2 above)
Comment (jeremie74) [reply apps/catalog/catalog/src/app/marketplace/title/avails/avails.component.ts:95]: I just tested it and it work, even if there is no bucket yet for the org

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**